packages/job-dispatch/src/job-creation.ts

```diff
@@ -88,62 +89,86 @@ export const createTriggeredReleaseJobs = async (
+ *
+ * @param je
+ */
+export const onJobCompletion = async (je: Job) => {
+  const triggers = await db
+    .select()
+    .from(releaseJobTrigger)
+    .innerJoin(release, eq(releaseJobTrigger.releaseId, release.id))
+    .innerJoin(environment, eq(releaseJobTrigger.environmentId, environment.id))
+    .innerJoin(deployment, eq(release.deploymentId, deployment.id))
+    .where(eq(releaseJobTrigger.jobId, je.id))
+    .then(takeFirst);
 
-    const affectedReleaseJobTriggers = await db
-      .select()
-      .from(releaseJobTrigger)
-      .leftJoin(job, eq(job.id, releaseJobTrigger.jobId))
-      .innerJoin(release, eq(releaseJobTrigger.releaseId, release.id))
-      .innerJoin(
-        environment,
-        eq(releaseJobTrigger.environmentId, environment.id),
-      )
-      .innerJoin(
-        environmentPolicy,
-        eq(environment.policyId, environmentPolicy.id),
-      )
-      .innerJoin(
-        environmentPolicyDeployment,
-        eq(environmentPolicyDeployment.policyId, environmentPolicy.id),
-      )
-      .where(
-        and(
-          isNull(releaseJobTrigger.jobId),
-          isNull(environment.deletedAt),
-          or(
-            and(
-              eq(releaseJobTrigger.releaseId, triggers.release.id),
-              eq(
-                environmentPolicyDeployment.environmentId,
-                triggers.environment.id,
-              ),
-            ),
-            and(
-              eq(environmentPolicy.releaseSequencing, "wait"),
-              eq(environment.id, triggers.environment.id),
+  const affectedReleaseJobTriggers = await db
+    .select()
+    .from(releaseJobTrigger)
+    .innerJoin(job, eq(releaseJobTrigger.jobId, job.id))
+    .innerJoin(environment, eq(releaseJobTrigger.environmentId, environment.id))
+    .leftJoin(environmentPolicy, eq(environment.policyId, environmentPolicy.id))
+    .leftJoin(
+      environmentPolicyDeployment,
+      eq(environmentPolicyDeployment.policyId, environmentPolicy.id),
+    )
+    .leftJoin(
+      releaseDependency,
+      and(
+        eq(releaseDependency.releaseId, releaseJobTrigger.releaseId),
+        eq(releaseDependency.deploymentId, triggers.deployment.id),
+      ),
+    )
+    .where(
+      and(
+        eq(job.status, "triggered"),
+        or(
+          // this release has a criteria requirement, i.e. "n from QA need to pass"
+          // and the completed job is part of the environment that needs to pass
+          and(
+            eq(releaseJobTrigger.releaseId, triggers.release.id),
+            eq(
+              environmentPolicyDeployment.environmentId,
+              triggers.environment.id,
             ),
           ),
+          // this release is waiting on previous releases in the same environment to finish,
+          // and the completed job is part of this environment
+          and(
+            eq(environmentPolicy.releaseSequencing, "wait"),
+            eq(environment.id, triggers.environment.id),
+          ),
```

Comment:
instead of doing inline comments, I would suggest extraing these into variables,


```ts
const areReleasesWaitingOnPreviousRelease = and(
    eq(environmentPolicy.releaseSequencing, "wait"),
    eq(environment.id, triggers.environment.id),
  )
  ```